test/framework/operator.go

```diff
@@ -105,92 +111,33 @@ var (
+func DeployClusterWideKogitoOperatorFromYaml(deploymentNamespace string) error {
+	yamlContent, err := ReadFromURI(config.GetOperatorYamlURI())
 	if err != nil {
-		return
-	}
-	created = append(created, cr)
-
-	crb := &rbac.ClusterRoleBinding{}
-	err = loadResource(namespace, deployURI+"clusterrole_binding.yaml", crb, func(object interface{}) {
-		// Object name needs to be unique so we can create independent objects for parallel tests
-		object.(*rbac.ClusterRoleBinding).SetName(object.(*rbac.ClusterRoleBinding).GetName() + "-" + namespace)
-		for i := range object.(*rbac.ClusterRoleBinding).Subjects {
-			object.(*rbac.ClusterRoleBinding).Subjects[i].Namespace = namespace
-		}
-		object.(*rbac.ClusterRoleBinding).RoleRef.Name = cr.Name
-	})
-	if err != nil {
-		return
+		GetLogger(deploymentNamespace).Error(err, "Error while reading kogito-operator.yaml file")
+		return err
 	}
-	created = append(created, crb)
 
-	if IsOpenshift() {
-		// Wait for docker pulling secret available for kogito-operator serviceaccount
-		// This is needed if images are stored into local Openshift registry
-		// Note that this is specific to Openshift
-		err = WaitForOnOpenshift(namespace, "image pulling secret", 2, func() (bool, error) {
-			// unfortunately the SecretList is buggy, so we have to fetch it manually: https://github.com/kubernetes-sigs/controller-runtime/issues/362
-			// so use direct command to look for specific secret
-			output, err := CreateCommand("oc", "get", "secrets", "-o", "name", "-n", namespace).WithLoggerContext(namespace).Execute()
-			if err != nil {
-				GetLogger(namespace).Error(err, "Error while trying to get secrets")
-				return false, err
-			}
-			GetLogger(namespace).Info(output)
-			return strings.Contains(output, "secret/"+kogitoOperatorPullImageSecretPrefix), nil
-		})
+	yamlContent = strings.ReplaceAll(yamlContent, "quay.io/kiegroup/kogito-cloud-operator:"+version.Version, getOperatorImageNameAndTag())
 
-		if err != nil {
-			return
-		}
+	tempFilePath, err := CreateTemporaryFile("kogito-operator*.yaml", yamlContent)
+	if err != nil {
+		GetLogger(deploymentNamespace).Error(err, "Error while storing adjusted YAML content to temporary file")
+		return err
 	}
 
-	// Then deploy operator
-	d := &coreapps.Deployment{}
-	err = loadResource(namespace, deployURI+"operator.yaml", d, func(object interface{}) {
-		GetLogger(namespace).Debug("Using operator image", "image", getOperatorImageNameAndTag())
-		object.(*coreapps.Deployment).Spec.Template.Spec.Containers[0].Image = getOperatorImageNameAndTag()
-
-		if namespaced {
-			// Set Kogito operator to watch only namespace where it is installed
-			container := object.(*coreapps.Deployment).Spec.Template.Spec.Containers[0]
-			for i := range container.Env {
-				if container.Env[i].Name == "WATCH_NAMESPACE" {
-					container.Env[i].Value = namespace
-				}
-			}
-		}
-	})
+	_, err = CreateCommand("oc", "apply", "-f", tempFilePath).WithLoggerContext(deploymentNamespace).Execute()
```

Comment:
Shouldn't we do a `replace` instead ?

Reply:
umm, replace seems to require existing resource, so it doesn't work